### PR TITLE
Equivariant flow examples

### PR DESCRIPTION
**Pull request by LeonKlein**
Add examples notebooks for

- equivariant kernel flows (nODEs)
- equivariant RNVP flows

### CODE REVIEW

File: bgflow/distribution/normal.py [hunk 1]
```diff
@@ -236,3 +236,26 @@ def dim(self):
 
     def __len__(self):
         return self._dim
+
+class MeanFreeNormalDistribution(Energy, Sampler):
+    """ Mean-free normal distribution. """
```
Comment (Olllom) [inline bgflow/distribution/normal.py:241]: Same here; just a basic API test that checks if the energy and sample methods work without errors.

File: bgflow/distribution/energy/multi_double_well_potential.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import torch
+
+
+__all__ = ["MultiDoubleWellPotential"]
+
+
+class MultiDoubleWellPotential(Energy):
+    
+    def __init__(self, dim, n_particles, a, b, c, offset):
+        super().__init__(dim)
```
Comment (Olllom) [inline bgflow/distribution/energy/multi_double_well_potential.py:12]: Since this is an energy that operates on multidimensional events, you should register its dimension with super().__init__(n_particles, spatial_dim)

File: bgflow/distribution/normal.py [hunk 3]
```diff
@@ -236,3 +236,26 @@ def dim(self):
...
+
+    def sample(self, n_samples, temperature=1.):
+        x = torch.Tensor(n_samples, self._n_particles, self._spacial_dims).normal_(mean= 0, std=self._std)
+        return self._remove_mean(x)
+
+    def _remove_mean(self, x):
+        #return x
+        x = x.view(-1, self._n_particles, self._spacial_dims)
+        x = x - torch.mean(x, dim=1, keepdim=True)
+        return x.view(-1, self.dim).cuda()
```
[the remark this answers is not in the document]
Comment (Olllom) [reply bgflow/distribution/normal.py:261]: ... but you would have to change the sample construction to `torch.Tensor(...).normal_(std=1.0) * self._std` in order for this to work.

File: bgflow/utils/geometry.py [hunk 4]
```diff
@@ -101,3 +101,30 @@ def distances_from_vectors(r, eps=1e-6):
...
+    -------
+    samples : PyTorch tensor
+        Mean-free positions of n_particles in n_dimensions.
+    """
+
+    shape = samples.shape
+    if isinstance(samples, torch.Tensor):
+        samples = samples.view(-1, n_particles, n_dimensions)
+        samples = samples - torch.mean(samples, dim=1, keepdim=True)
+        samples = samples.view(*shape)
```
[the remark this answers is not in the document]
Comment (LeonKlein) [reply bgflow/utils/geometry.py:125]: Sure, but what about the different keywords for torch and numpy, i.e. `axis` vs `dim` and `keepdims` vs `keepdim`?

[the remark this answers is not in the document]
Comment (Olllom) [reply bgflow/utils/geometry.py:125]: ah, good point. Sorry, I missed that.

File: bgflow/utils/geometry.py [hunk 5]
```diff
@@ -11,7 +11,7 @@ def distance_vectors(x, remove_diagonal=True):
 
     Parameters
     ----------
-    x : PyTorch tensor
+    x : torch.tensor
```
Comment (Olllom) [inline bgflow/utils/geometry.py:14]: the class name is `torch.Tensor`

File: bgflow/utils/geometry.py [hunk 6]
```diff
@@ -20,7 +20,7 @@ def distance_vectors(x, remove_diagonal=True):
 
     Returns
     -------
-    r : PyTorch tensor
+    r : torch.tensor
```
Comment (Olllom) [inline bgflow/utils/geometry.py:23]: See above

**Review by Olllom — APPROVED** [review]
just one picky detail about the docstrings. otherwise LGTM

**Review by Olllom — APPROVED** [review]
Nice! Thanks
